### PR TITLE
Add comments and suggest alt type names for newbie clarity (server only)

### PR DESCRIPTION
**Pull request by bentayloruk**
I'm learning Fable and Elmish and using this wonderful scaffold.  I've worked through the server code, trying to understand it and then commenting it.  As a newbie myself, I thought the comments might be helpful for other newbies that come to the code.  Then again, maybe not, comments are controversial these days!

I've also renamed a few types for clarity (IMO).  Maybe you will like them, but they are in a separate commit, so easy to drop.

I've also made a couple of functions private, just to limit the module surface area.  

If this looks helpful, I'll do the client code too.

### CODE REVIEW

File: src/Server/Auth.fs [hunk 1]
```diff
@@ -1,35 +1,41 @@
...
+(*  Login web part and functions for API web part request authorisation with JWT *)
+
 open Suave
 open Suave.RequestErrors
 
 let unauthorized s = Suave.Response.response HTTP_401 s
 
 let UNAUTHORIZED s = unauthorized (UTF8.bytes s)
 
+// Login web part that authenticates a user and returns a token in the HTTP body.
```
Comment (dsyme) [inline src/Server/Auth.fs]: Use /// comments

File: src/Server/FableJson.fs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+module ServerCode.FableJson
+
+(*
```
Comment (dsyme) [inline src/Server/FableJson.fs]: Don't use ``(* ... *)`` comments - just use ``//`` or ``///``

**Comment by dsyme** [comment]
@bentayloruk Really nice to have these kinds of improvements

**Comment by bentayloruk** [comment]
@dsyme thanks for the comment schooling Don!  Time for the blush face...  😊   I shall fix that up.

**Comment by forki** [comment]
that lloks very cool. sorry for merge conflicts. please rebase

**Comment by bentayloruk** [comment]
Sorry about that.  I've rebased and updated dependencies.  Looks like a bump in Suave since the update yesterday.  I do have a test failure at my end, with `login with test user`.  Possible it's a timeout thing as that is mentioned in the message.  Will look into it.

I'll remove WIP, as happy to have the server changes merged if you like them.  I will have a dabble with the client side too.

**Comment by forki** [comment]
can you please rebase? thx

**Comment by bentayloruk** [comment]
Sure thing.  I've done the rebase, but seem to be having an issue related to the port change (login page fails for me).  Will try and get to the bottom of it and then update the PR.

**Comment by bentayloruk** [comment]
All sorted.  I ran `.paket\paket.exe update` again and there were some dep updates which I squashed into the previous deps commit I did on Friday.  Should I be doing that, or leaving the Paket updates to you/someone on the core team?

**Comment by forki** [comment]
update all the things!

**Pull request merged**